adapter/slack.py

```diff
@@ -64,12 +83,18 @@ async def ask(self, request: Request):
 
             bot_engine = self.engine_selector.select_engine(engine_type=chatbot.model)
 
-            chatbot_response = bot_engine.generate_response(query=question)
-
-            self.app.client.chat_update(
-                channel=channel_id, ts=loading_message["ts"], text=chatbot_response
+            send_response_thread = threading.Thread(
+                target=self.send_generated_response,
+                kwargs={
+                    "channel": channel_id,
+                    "ts": loading_message["ts"],
+                    "engine": bot_engine,
+                    "question": question,
+                },
             )
 
+            send_response_thread.start()
+
             return Response(status_code=200)

```

Comment:
So it will return a success even before the bot finishes generating the message?

Reply:
The success here is not a successful response in the traditional sense but an acknowledgement. Slack requires this or else it will think an error/timeout happened. You can read more about it [here](https://api.slack.com/interactivity/slash-commands) and [here](https://api.slack.com/interactivity/handling#async_responses).